drf_roles/mixins.py

```diff
@@ -19,7 +18,14 @@ class RoleError(Exception):
 class RoleViewSetMixin(object):
     """A ViewSet mixin that parameterizes DRF methods over roles"""
     _viewset_method_registry = set(getattr(settings, "VIEWSET_METHOD_REGISTRY", DEFAULT_REGISTRY))
-    _role_groups = set(getattr(settings, "ROLE_GROUPS", DEFAULT_GROUPS))
+
+    def __init__(self, **_kwargs):
+        groups = []
+        if hasattr(settings, "ROLE_GROUPS"):
+            groups = set(getattr(settings, "ROLE_GROUPS", groups))
+        else:
+            groups = [group.name.lower() for group in Group.objects.all()]
+        self._role_groups = set(groups)
```

Comment:
I would just inline `groups` here like:

```python
self._role_groups = set(getattr(settings, "ROLE_GROUPS", [group.name.lower() for group in Group.objects.all()]))
```

Reply:
Good call: 24c54fee625577301d2205c107fcfabdabaf246e

Reply:
The side-effect of this is that there is always a read on the `auth.group` table.  The original approach avoids that extra hit.

Reply:
Ah I see, because the arguments to getattr are evaluated 👍